### PR TITLE
Add start screen and controls menu.

### PR DESCRIPTION
**Pull request by tclaar**
Add a number of entities for the start and controls menus.
Also fix the pause and respawn buttons not working.

### CODE REVIEW

File: js/classes/entities/foreground/ui/button.js
Comment (calebkrauter) [inline js/classes/entities/foreground/ui/button.js:1]: Thank you for adding this.

File: js/classes/entities/foreground/ui/button.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+     * Constructor for a Button.
+     * 
+     * @param {Vector} pos the position, in pixels, of the Button on the canvas
+     * @param {Vector} size the size, in pixels, of the Button on the canvas
+     * @param {string} text the text displayed on the Button
+     * @param {number} fontSize the font size of the Button's text
+     * @param {function} onClick the function to call when the button is clicked
+     * @param {boolean} [selfDestruct] whether or not the Button should destroy itself once clicked
+     */
+    constructor(pos, size, text, fontSize, onClick, selfDestruct = true) {
```
Comment (NathanHinthorne) [inline js/classes/entities/foreground/ui/button.js:17]: Good name for the selfDestruct parameter :)

File: js/classes/entities/foreground/ui/button.js
Comment (calebkrauter) [reply js/classes/entities/foreground/ui/button.js:1]: Super cool.

File: js/classes/entities/foreground/ui/controls.js
Comment (calebkrauter) [inline js/classes/entities/foreground/ui/controls.js:1]: Thank you for adding this in!

File: js/classes/entities/midground/weapon/slingshot.js [hunk 2]
```diff
@@ -74,7 +74,7 @@ class Slingshot {
         // this.chadCenter = Vector.add(CHAD.pos, new Vector(CHAD.scaledSize.x / 2, CHAD.scaledSize.y / 2));
         this.timeSinceLastShot += GAME.clockTick;
 
-        if (!HUD.pauseButton.isMouseOver() && CHAD.health > 0) {
+        if ((!HUD.pauseButton || !HUD.pauseButton.isMouseOver()) && CHAD.health > 0) {
```
Comment (NathanHinthorne) [inline js/classes/entities/midground/weapon/slingshot.js:77]: Thanks for fixing this!

File: js/classes/entities/foreground/ui/controls.js
Comment (NathanHinthorne) [inline js/classes/entities/foreground/ui/controls.js:1]: We really needed this. Thanks for adding it @tclaar !

**Review by NathanHinthorne — APPROVED** [review]

File: js/classes/entities/foreground/ui/hud.js [hunk 3]
```diff
@@ -172,21 +173,7 @@ class Hud {
      * Update the HUD.
      */
     update() {
-        // if the user clicks on the pause/play button, toggle GAME.running
```
Comment (calebkrauter) [inline js/classes/entities/foreground/ui/hud.js:153]: So update does nothing now? Is that intentional?

File: js/classes/game_engine.js [hunk 4]
```diff
@@ -65,6 +65,22 @@ class GameEngine {
         return 0.25;
     }
 
+    /** The current mode of the GameEngine. */
+    get mode() {
+        return this._mode;
```
Comment (calebkrauter) [inline js/classes/game_engine.js:70]: This is a setter right? It feels weird to return a setter if that is what's happening.

**Review by calebkrauter — APPROVED** [review]
Looks great.

File: js/classes/entities/foreground/ui/hud.js (hunk 3 above)
Comment (tclaar) [reply js/classes/entities/foreground/ui/hud.js:153]: Yeah, this behavior was moved into the `PauseButton` component.

**Pull request merged**